wordclock/sntp_client.c

```diff
@@ -67,24 +56,25 @@ void sntp_tsk(void *pvParameters)
 		uint32_t t = RTC.COUNTER;
 		vTaskDelayMs(delaySec * 1000);
 		_rtcTicsPerSec = (RTC.COUNTER - t) / delaySec;		
-//		time_t ts = time(NULL);
-//		printf("TIME: %s", ctime(&ts));
+		time_t ts = time(NULL);
+		printf("TIME: %s", ctime(&ts));
 	}
 }
 
-bool sntp_client_time_valid() {
+bool sntp_client_time_valid(void) {
 	if (_rtcTicsPerSec == 0) return false;
 
-	// sntp_fun.c stores the last update in RTC scratch2 register. See line:
-	//     tim_ref = now_rtc;
-	// At the bottom of the c file.
-	// Use that to calculate the age of the time update.
-	uint32_t age = (RTC.COUNTER - RTC.SCRATCH[2]) / _rtcTicsPerSec;
+	uint32_t age = time(NULL) - sntp_last_update_ts();
 	return (age <= (2 * UPDATE_INERVAL) / 1000)? true : false;
 }
 
-void sntpClientIinit(const struct timezone* tz)
+void sntp_client_init(void)
 {
-	_tz = *tz;
-     xTaskCreate(sntp_tsk, "SNTP", 1024, NULL, 1, NULL);
+	printf("Starting SNTP... \n");
+
+	sntp_set_update_delay(UPDATE_INERVAL);
+	sntp_set_servers(servers, sizeof(servers) / sizeof(char*));
+	sntp_initialize(&_tz);
+    
+	xTaskCreate(sntp_tsk, "SNTP", 256, NULL, 1, NULL);
```

Comment:
Heb je de stacksizes van de taken gecheckt? Was 1024, nu 256

Reply:
Ik loop tegen geen problemen aan. Er gebeurd namelijk niks meer in sntp_task, er gebeurd nog meer in evenTask.